gcc/rust/ast/rust-expr.h

```diff
@@ -4234,10 +4235,11 @@ struct MatchArm
   // Constructor for match arm with a guard expression
   MatchArm (std::vector<std::unique_ptr<Pattern> > match_arm_patterns,
 	    std::unique_ptr<Expr> guard_expr = nullptr,
-	    std::vector<Attribute> outer_attrs = std::vector<Attribute> ())
+	    std::vector<Attribute> outer_attrs = std::vector<Attribute> (),
+	    Location locus = Location ())
```

Comment:
I think your going to have to update this to be a mandatory arguments just like the HIR so you will need to update the code in rust-parse-impl.h to actually grab the location info it will be fine to ignore attributes for now.